### PR TITLE
Touch pan

### PR DESCRIPTION
**Pull request by ralucaportase**
* Ignore touch outside canvas
* If distance between touches is constant, pan. After the distance is big enough,  start zooming no matter how small or big the difference between touches becomes.
* If multiple touches exist, consider just the first two of them. The touches are sorted based on the moment the screen was touched

### CODE REVIEW

File: jquery.flot.touchNavigate.js [hunk 1]
```diff
@@ -168,16 +169,21 @@
...
+            // send the computed touched axis to the zoom function so that it only zooms on that one
+            plot.zoom({
+                center: center,
+                amount: zoomAmount,
+                axes: navigationState.touchedAxis
+            });
+            gestureState.prevDistance = dist;
+
+            //activate zomm mode
+            gestureState.twoTouches = true;
```
Comment (cristiingineru) [inline jquery.flot.touchNavigate.js]: This assignment should be moved to pinch.drag() as well.

The name is a bit confusing since the pinch.drag() is called when 2 touches are pressed. Maybe zoomEnabled?

File: jquery.flot.uiConstants.js [hunk 2]
```diff
@@ -4,6 +4,7 @@
         SNAPPING_CONSTANT: 20,
         PANHINT_LENGTH_CONSTANT: 10,
         MINOR_TICKS_COUNT_CONSTANT: 4,
-        TICK_LENGTH_CONSTANT: 10
+        TICK_LENGTH_CONSTANT: 10,
+        DISTANCE_MARGIN: 25
```
Comment (cristiingineru) [inline jquery.flot.uiConstants.js]: The name should include ZOOM since it's used by the zoom/pinch.

File: tests/jquery.flot.touchNavigate.Test.js [hunk 3]
```diff
@@ -67,7 +67,50 @@ describe("flot touch navigate plugin", function () {
             expect(xaxis.max).toBeCloseTo(initialXmax - (initialXmax - midPointCoords.x) * (1 - 1/amount), 6);
             expect(yaxis.min).toBeCloseTo((midPointCoords.y - initialYmin) * (1 - 1/amount) + initialYmin, 6);
             expect(yaxis.max).toBeCloseTo(initialYmax - (initialYmax - midPointCoords.y) * (1 - 1/amount), 6);
-          });
+        });
+
+        it('for multiple touches should take into account just the first two of them and act acordingly',function() {
```
Comment (cristiingineru) [inline tests/jquery.flot.touchNavigate.Test.js]: Dont' forget to add a test for the case when everything starts correctly with 2 touches but a 3rd one interfere.

File: tests/jquery.flot.touchNavigate.Test.js [hunk 4]
```diff
@@ -315,6 +358,108 @@ describe("flot touch navigate plugin", function () {
 
       });
 
+      it('should pan the plot for two fingers with constant distance',function() {
```
Comment (cristiingineru) [inline tests/jquery.flot.touchNavigate.Test.js]: We need to clean up things here. The touchNavigate plugin deals with high level messages like: pan, pinch and double tap. The describe sections of this file should be named accordingly.

This test is testing panning but as a corner case for pinch, so it should be moved in the pinch section.

There should be an almost identical test which makes sure no zoom is happening even when there is a small variation in distance.

File: jquery.flot.touch.js [hunk 5]
```diff
@@ -247,7 +247,13 @@
         }
 
         function isPinchEvent(e) {
-            return e.touches && e.touches.length === 2;
+            if (e.touches && e.touches.length >= 2) {
```
Comment (cristiingineru) [inline jquery.flot.touch.js]: Not necessarily for now, but some new tests have to be added for this code. Unfortunate there are only a few tests here for this plugin:
https://github.com/ni-kismet/engineering-flot/blob/master/tests/jquery.flot.touch.Test.js
